src/main/java/sggw/wzim/czasnawypad/service/AttractionRatingService.java

```diff
@@ -0,0 +1,79 @@
+package sggw.wzim.czasnawypad.service;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import sggw.wzim.czasnawypad.db.AttractionRepository;
+import sggw.wzim.czasnawypad.db.dto.AttractionRatingDTO;
+import sggw.wzim.czasnawypad.db.dto.CreateAttractionRatingDTO;
+import sggw.wzim.czasnawypad.db.entity.Attraction;
+import sggw.wzim.czasnawypad.db.entity.AttractionRating;
+import sggw.wzim.czasnawypad.db.entity.User;
+import sggw.wzim.czasnawypad.repository.AttractionRatingRepository;
+
+@Service
+@RequiredArgsConstructor
+public class AttractionRatingService {
+    private final AttractionRatingRepository ratingRepository;
+    private final AttractionRepository attractionRepository;
+
+    public List<AttractionRatingDTO> getRatingsForAttraction(Integer attractionId) {
+        List<AttractionRating> ratings = ratingRepository.findByAttractionId(attractionId);
+        return ratings.stream()
+                      .map(this::toDto)
+                      .collect(Collectors.toList());
+    }
+    
+    public List<AttractionRatingDTO> getRatingsByUser(User user) {
+        return ratingRepository.findByUser(user).stream().map(this::toDto).collect(Collectors.toList());
+    }
+
+    public AttractionRatingDTO addRating(User user, CreateAttractionRatingDTO dto) {
+        Attraction attraction = attractionRepository.findById(dto.getAttractionId())
+                .orElseThrow(() -> new IllegalArgumentException("Attraction not found"));
```

Comment:
Please define dedicated exception like `AttractionNotFoundException` and add it's handling to `src/main/java/sggw/wzim/czasnawypad/controller/RestExceptionHandler.java`